M365/MDO/MDOThreatPolicyChecker.ps1

```diff
@@ -0,0 +1,926 @@
+                (($_.Value.GetType() -eq [Boolean] -and $_.Value -eq $true) `
+                        -or ($_.Value -ne '{}' -and $_.Value -ne 'Off' -and $_.Value -ne $true -and $_.Value -ne '' -and $excludedProperties -notcontains $_.Name))) {
+                Write-Host "`t`t$($_.Name): $($_.Value)"
+            } else {
+                Write-Verbose "`t`tExcluded property:$($_.Name): $($_.Value)"
+            }
+        }
+        Write-Host " "
+    }
+
+    function Get-Policy {
+        param(
+            $Rule = $null,
+            $PolicyType = $null
+        )
+
+        if ($null -eq $Rule) {
+            if ($PolicyType -eq "Anti-phish") {
+                $policyDetails = "`n$PolicyType (Impersonation, Mailbox/Spoof Intelligence, Honor DMARC):`n`tThe Default policy."
+            } elseif ($PolicyType -eq "Anti-spam") {
+                $policyDetails = "`n$PolicyType (includes phish & bulk actions):`n`tThe Default policy."
+            } else {
+                $policyDetails = "`n${PolicyType}:`n`tThe Default policy."
+            }
+        } else {
+            if ($PolicyType -eq "Anti-phish") {
+                $policyDetails = "`n$PolicyType (Impersonation, Mailbox/Spoof Intelligence, Honor DMARC):`n`tName: {0}`n`tPriority: {1}" -f $Rule.Name, $Rule.Priority
+            } elseif ($PolicyType -eq "Anti-spam") {
+                $policyDetails = "`n$PolicyType (includes phish & bulk actions):`n`tName: {0}`n`tPriority: {1}" -f $Rule.Name, $Rule.Priority
+            } else {
+                $policyDetails = "`n${PolicyType}:`n`tName: {0}`n`tPriority: {1}" -f $Rule.Name, $Rule.Priority
+            }
+        }
+        return $policyDetails
+    }
+
+    function Test-GraphContext {
+        [OutputType([bool])]
+        param (
+            [Parameter(Mandatory = $true)]
+            [string[]]$Scopes,
+            [Parameter(Mandatory = $true)]
+            [string[]]$ExpectedScopes
+        )
+
+        $validScope = $true
+        foreach ($expectedScope in $ExpectedScopes) {
+            if ($Scopes -contains $expectedScope) {
+                Write-Verbose "Scopes $expectedScope is present."
+            } else {
+                Write-Host "The following scope is missing: $expectedScope" -ForegroundColor Red
+                $validScope = $false
+            }
+        }
+        return $validScope
+    }
+
+    function Write-DebugLog ($message) {
+        if (![string]::IsNullOrEmpty($message)) {
+            $Script:DebugLogger = $Script:DebugLogger | Write-LoggerInstance $message
+        }
+    }
+
+    function Write-HostLog ($message) {
+        if (![string]::IsNullOrEmpty($message)) {
+            $Script:HostLogger = $Script:HostLogger | Write-LoggerInstance $message
+        }
+        # all write-host should be logged in the debug log as well.
+        Write-DebugLog $message
+    }
+
+    Import-Module Microsoft.Graph.Authentication
+    Import-Module ExchangeOnlineManagement
```

Comment:
We have a bug if you import ExchangeOnlineManagement before Microsoft.Graph.Authentication and that is the reason to put in that order.
Additionally, if we publish the other PR, about shared functions to connect M365, we will remove #Requires.

Reply:
But doesn't the `#Requires` handle importing the module right away?

Reply:
yes, we should remove it if you can test that scenario out. 

New PowerShell session run the script. It stops because you are not connected to EXO session. Then you connect, then run the script again. Does it work correctly as you would expect without needing this import cmdlet. If it does, then yes we should remove it as it isn't relevant to this PR yet.

Reply:
I tested before and I am not wrong it works fine, but I will double check.

Reply:
I just wanted to make sure it was checked without the two `Import-Module` there since you provided there was a bug that required modules to be loaded in a particular order. As I have no way to test this out currently.

Reply:
It worked with the current #requires  
You have more details in this request;
https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/2803
seems to fail with other products as well.